### PR TITLE
Fix `CreateOrUpdateOrgSecret` regression introduced in v53

### PR DESCRIPTION
**Pull request by frankywahl**
`CreateOrUpdateOrgSecret`'s API uses `string` for selected repository IDS while other APIs use `int`. This fixes the issue locally by implementing its own type.

fixes https://github.com/google/go-github/issues/2816

### CODE REVIEW

File: github/dependabot_secrets.go
Comment (frankywahl) [inline github/dependabot_secrets.go:167]: Created this inline for now as it's the only place it's used. Happy to pull it out as an unexported function to match the `putSecret` package function (maybe `putOrgSecret`). That would require though defining a package type for `params` (for used as an input argument). I thought it was an overkill for now, so left it as such.

Comment (gmlewis) [reply github/dependabot_secrets.go:167]: There's no need for the inline function here.  Let's just please delete lines 160 and 167, and move lines 161-166 down to replace line 170 below.

File: github/dependabot_secrets.go
Comment (gmlewis) [inline github/dependabot_secrets.go:170]:
```suggestion
	req, err := s.client.NewRequest("PUT", url, params)
	if err != nil {
		return nil, err
	}

	return s.client.Do(ctx, req, nil)
```

File: github/dependabot_secrets.go
Comment (gmlewis) [inline github/dependabot_secrets.go:150]:
```suggestion
		repoIDs[i] = fmt.Sprintf("%v", secret)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @frankywahl!
Just a few tweaks, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

File: github/dependabot_secrets.go
Comment (frankywahl) [reply github/dependabot_secrets.go]: Sounds good - was trying to keep the symmetry with what was previously there. Removed!

File: github/dependabot_secrets.go
Comment (gmlewis) [reply github/dependabot_secrets.go]: No, let's please change it back to `%v` because it is idiomatic Go to use `%v` unless there is a specific reason otherwise, such as if we need leading zeros, as in `%05d`.
Although you may be able to find other examples in this repo where this is a `%d`, we have attempted to remain consistent and use `%v` unless there is a specific need.  Thanks.

**Comment by gmlewis** [comment]
Also, in the future, please don't use force-push in PRs to this repo, as it makes reviewing more challenging to see what has changed since the last review.
See CONTRIBUTING.md for more details.  Thanks.

**Review by gmlewis — APPROVED** [review]
Thank you, @frankywahl !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
